scripts/seed-database.js

```diff
@@ -0,0 +1,61 @@
+import { createConnection } from 'mysql2/promise'
+import bcrypt from 'bcrypt'
+
+async function seed () {
+  const connection = await createConnection({
+    multipleStatements: true,
+    host: process.env.MYSQL_HOST,
+    port: Number(process.env.MYSQL_PORT),
+    database: process.env.MYSQL_DATABASE,
+    user: process.env.MYSQL_USER,
+    password: process.env.MYSQL_PASSWORD
+  })
+
+  try {
+    await truncateTables(connection)
+    await seedUsers(connection)
+
+    /* c8 ignore start */
+  } catch (error) {
+    console.error('Error seeding database:', error)
+  } finally {
+    /* c8 ignore end */
+    await connection.end()
+  }
+}
+
+async function truncateTables (connection) {
+  const [tables] = await connection.query('SHOW TABLES')
+
+  if (tables.length > 0) {
+    const tableNames = tables.map((row) => row[`Tables_in_${process.env.MYSQL_DATABASE}`])
+    const truncateQueries = tableNames.map((tableName) => `TRUNCATE TABLE \`${tableName}\``).join('; ')
+
+    await connection.query('SET FOREIGN_KEY_CHECKS = 0')
+    try {
+      await connection.query(truncateQueries)
+      console.log('All tables have been truncated successfully.')
+    } finally {
+      await connection.query('SET FOREIGN_KEY_CHECKS = 1')
+    }
+  }
+}
+
+async function seedUsers (connection) {
+  const usernames = ['basic', 'moderator', 'admin']
+
+  for (const username of usernames) {
+    const hash = await bcrypt.hash('password', 10)
```

Comment:
Use `scrypt` instead, so you don't need external dependency.
See https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#introduction

Reply:
Can you provide (or give a link to) what you consider a safe implementation please?

Reply:
No link provided, but it is what I currently using with some modify.
For what I use, I store also the scrypt parameter, so I can migrate the old user to safer constant when time pass.
Here is using constant for simple usage.

I choose the second config, with 2 parallelism.
https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#scrypt

```js
import { scrypt, timingSafeEqual, randomBytes } from 'crypto'

// Scrypt constants
const SCRYPT_KEYLEN = 32
const SCRYPT_COST = 65536
const SCRYPT_BLOCK_SIZE = 8
const SCRYPT_PARALLELIZATION = 2
const SCRYPT_MAXMEM = 128 * SCRYPT_COST * SCRYPT_BLOCK_SIZE * 2

function hash(value) {
  // Salt must be at least 16 bytes long
  const salt = randomBytes(Math.min(16, SCRYPT_KEYLEN / 2))
  return new Promise((resolve, reject) => {
    scrypt(value, salt, SCRYPT_KEYLEN, { 
      cost: SCRYPT_COST,
      blockSize: SCRYPT_BLOCK_SIZE,
      parallelization: SCRYPT_PARALLELIZATION,
      maxmem: SCRYPT_MAXMEM
    }, function (error, key) {
      if (error !== null) reject(error)
      resolve(`${salt.toString('hex')}.${key.toString('hex')}`)
    })
  })
}

function compare(value, hash) {
  const [salt, hashed] = hash.split('.')
  const buf = Buffer.from(hashed, 'hex')
  new Promise((resolve) => {
    scrypt(value, salt, SCRYPT_KEYLEN, { 
      cost: SCRYPT_COST,
      blockSize: SCRYPT_BLOCK_SIZE,
      parallelization: SCRYPT_PARALLELIZATION,
      maxmem: SCRYPT_MAXMEM
    }, function (error, key) {
      if (error !== null) {
        timingSafeEqual(buf, buf)
        resolve(false)
      } else {
        resolve(timingSafeEqual(key, buf))
      }
    })
  })
}
```

Reply:
Interesting, 

I think it would be worthwhile to create an opensource plugin for this.

Reply:
Can you take a look? I am not big on encryption stuff: https://github.com/fastify/demo/pull/1/commits/776086b5c5c7c877067a8488abafb2f2210b18c0